File: python/randombot.py
import random

from api import ElevatorSimulation, DIRECTIONS, ACTIONS


def random_bot():
    """A simple bot that will randomly move and stop"""
    simulation = ElevatorSimulation(
        event="secondspace2025",
        building_name="tiny_random",
        bot="random",
        email="bob@mail.com",
    )
    current_state = simulation.initial_state
    while current_state["running"]:
        commands = [
            {
                "elevator_id": elevator["id"],
                "direction": random.choice(DIRECTIONS),
                "action": random.choice(ACTIONS),
            }
            for elevator in current_state["elevators"]
        ]
        current_state = simulation.send(commands)
    print("Score:", current_state.get("score"))
    print("Replay URL:", current_state.get("replay_url"))


if __name__ == "__main__":
    random_bot()

File: typescript/randomBot.ts
import { Command, Simulation } from "./api";
import { DIRECTIONS, ACTIONS } from "./constants";
import { BuildingName } from "./types";

async function randomBot(): Promise<void> {
  const simulation = new Simulation(
    "secondspace2025",
    BuildingName.TINY_RANDOM,
    "random-ts-bot",
    "bob@mail.com",
    true
  );
  await simulation.init();

  let currentState = simulation.initialState;
  while (currentState.running) {
    const commands = currentState.elevators.map(
      (elevator) =>
        new Command(
          elevator.id,
          Math.random() > 0.5 ? DIRECTIONS.UP : DIRECTIONS.DOWN,
          Math.random() > 0.5 ? ACTIONS.MOVE : ACTIONS.STOP
        )
    );
    currentState = await simulation.send(commands);
  }

  console.log("Score:", currentState.score);
  console.log("Replay URL:", currentState.replay_url);
}

randomBot();
